packages/xrpl/src/models/methods/tx.ts

```diff
@@ -11,7 +11,14 @@ import { BaseRequest, BaseResponse } from './baseMethod'
  */
 export interface TxRequest extends BaseRequest {
   command: 'tx'
-  transaction: string
+  /**
+   * The transaction hash to look up. Exactly one of `transaction` or `ctid` must be specified for a TxRequest.
+   */
+  transaction?: string
+  /**
+   * The Concise Transaction ID to look up. Exactly one of `transaction` or `ctid` must be specified for a TxRequest.
+   */
+  ctid?: string
```

Comment:
This should be a more specific type to ensure that one of the two is provided.

Reply:
From what I saw when looking into this, I couldn't find a good way to ensure that one of the two was provided in a backwards compatible way. 

The normal way "one of" is written is by combining multiple discrete types into a type alias, but then you can't initialize it like we do in snippets / a lot of code with `Tx(...)`. So I don't think that path is worthwhile. 

That problem is the same one with doing typing with meta types like this: https://stackoverflow.com/questions/40510611/typescript-interface-require-one-of-two-properties-to-exist

Any thoughts on how to ensure only one of the two is specified?

Reply:
One or both will be returned.  CTID does not remove hash.